src/components/settingsForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import { User } from "./profileCard";
+
+interface UserFormFieldsProps {
+  user: User
+  firstNameRef: React.Ref<HTMLInputElement>;
+  lastNameRef: React.Ref<HTMLInputElement>;
+  aboutRef: React.Ref<HTMLInputElement>;
+  locationRef: React.Ref<HTMLInputElement>;
+}
+
+const SettingsForm: React.FC<UserFormFieldsProps> = ({
+  user,
+  firstNameRef,
+  lastNameRef,
+  aboutRef,
+  locationRef,
+}) => {
+  return (
+    <>
+      <div>
+        <label
+          htmlFor="firstname"
+          className="block text-lg font-semibold text-gray-700"
+        >
+          First Name
+        </label>
+        <input
+          id="firstname"
+          type="text"
+          ref={firstNameRef}
+          defaultValue={user.first_name ?? ""}
+          className="mt-1 w-full border border-slate-300 rounded-md px-3 py-2 focus:outline-none focus:ring-2 focus:ring-blue-500"
+        />
+      </div>
```

Comment:
make `InputField` component that accepts a `label`/`name`/`ref`/`defaultValue`/`type` as props (maybe more?)